### ISSUE TITLE
Bilder Taggen ähnlich von rep0st nur anhand von den Benis

### ISSUE THREAD

**Issue by GitSimaxe** [Scarwolf/p0weruser#102]
## Allgemeine Informationen
**Browser:** Alle

**Version:** Alle

**Modul:** rep0st

## Beschreibung

Wäre es möglich einen festen Filter zu implementieren ähnlich der Lesezeichen die man sich selbst erstellen kann, nur eben global egal wo man sich befindet. Sprich dass man Bilder mit einem gewissen negativen Benis Wert markiert oder gar ausblendet. Ich bin es leid ständig schrottige Bilder/Videos zu sehen. Aber immer wieder auf die Lesezeichen klicken nervt auch.

In den Einstellungen dann z.B ein Feld zu haben um z.b. -10 als minimal Wert festzulegen und alles was -10 oder mehr minus hat wird getaggt oder am besten gleich ausgeblendet.

**Comment by twobiers** [comment]
Ja, könnte man machen. Muss ich mir mal Gedanken machen, wie man das sinnvoll integrieren kann.

Stand heute kannst du das DefaultFilters / Standard Filter Modul aktivieren und dann mit dem [erweiterten Suchsyntax](https://pr0gramm.com/new/2782197) entsprechend Filter erstellen, die beim Aufruf der Seite greifen. Bspw ` s:100` zeigt dann alle Bilder mit Benis >= 100 an. 
Es ist leider nicht trivial sowas zu implementieren ohne Features einzuschränken, deshalb greift dieser Filter **nur beim ersten Seitenaufruf**. Das wäre also aktuell ein möglicher Workaround.

**Comment by GitSimaxe** [comment]
den s:100 habe ich auch schon benutzt, nur leider kann man diesen nicht runter setzen, also z.B s:0 oder s:1 und alles was mehr hat. genauso wie s:shit welcher ja erst ab -300 fest funktioniert. 

wenn ich die default filter in den Settings aktiviere kann ich ja für alle drei Globalen Filter (SWF, NSFW , NSFL) jeweils unterschiedliche Suchfilter setzen die dann aber aneinander gesetzt werden. Da ich meistens "Alle" globalen Filter anhabe, kaskadieren dadurch alle Filter zu einem. Sprich ich darf nur in einem etwas eintragen, da es sonst doppelt/dreifach vorkommt. (was du vermutlich meintest mit, dass es andere Features dann einschränkt.)

Ich werde mal solange versuchen die idealen Filter zu setzen, der den meisten "müll" aussortiert.

Danke aber schon einmal.

**Comment by twobiers** [comment]
Es ist halt leider sehr komplex so ein "Standard-Verhalten" abzubilden, deswegen ist das Konzept der Lesezeichen schon sinnvoller.
Mal angenommen wir würden einen Benis-Filter einbauen, was würde in den Folgenden Fällen passieren?

- Nutzer wechselt Filter
- Nutzer wechselt von Beliebt auf Neu
- Nutzer sieht sich Uploads eines anderen Nutzers (oder sich selbst) an
- Nutzer sieht sich Sammlungen an
- Nutzer sucht explizit über `s:` nach Posts außerhalb des Filters

In machen Fällen **kann** es sinnvoll sein den Filter anzuwenden, in manchen Fällen macht es keinen Sinn und geht gegen den eigentlichen Willen des Nutzers. 

Als UserScript können wir auch nicht so leicht Posts beim betrachten "verstecken", deswegen ist es notwendig auf die native Suche zurückzugreifen. Das einzige was ich mir aus der Hüfte geschossen sinnvoll vorstellen kann ist (ähnlich wie bei rep0st, FilterMark) der Fall eine Markierung des Posts im Stream.
